codypy/server.py

```diff
@@ -1,31 +1,35 @@
 import asyncio
+import logging
 import os
-import sys
 from asyncio.subprocess import Process
 
-from codypy.config import RED, RESET, YELLOW
-from codypy.logger import log_message
+from codypy.exceptions import (
+    AgentBinaryDownloadError,
+    AgentBinaryNotFoundError,
+    ServerTCPConnectionError,
+)
 from codypy.messaging import _send_jsonrpc_request
 from codypy.utils import (
     _check_for_binary_file,
     _download_binary_to_path,
     _format_binary_name,
 )
 
+logger = logging.getLogger(__name__)
+
 
 async def _get_cody_binary(binary_path: str, version: str) -> str:
     has_agent_binary = await _check_for_binary_file(binary_path, "cody-agent", version)
     if not has_agent_binary:
-        print(
-            f"{YELLOW}WARNING: The Cody Agent binary does not exist at the specified path: {binary_path}{RESET}"
+        logger.warning(
+            "Cody Agent binary does not exist at the specified path: %s", binary_path
         )
-        print(f"{YELLOW}WARNING: Start downloading the Cody Agent binary...{RESET}")
+        logger.warning("Start downloading the Cody Agent binary")
         is_completed = await _download_binary_to_path(
             binary_path, "cody-agent", version
         )
         if not is_completed:
-            print(f"{RED}ERROR: Failed to download the Cody Agent binary.{RESET}")
-            sys.exit(1)
+            raise AgentBinaryDownloadError("Failed to download the Cody Agent binary")
```

Comment:
We have to be careful by throwing if anything happens with downloading or network issues.
Since codypy wrapper is meant as a library by plugging in into a possible application e.g Sublime, throwing might not be the best option overall. Usually this should be handled by the application itself.
`sys.exit(1)` was also not the best solution here too.

Reply:
I think the download functionality should be outside the code library. Once you embed the core into an application e.g. a sublime plugin, this can be useful, but otherwise the core should not deal with this besides implementing APIs to talk to Cody.